app/components/list/index.tsx

```diff
@@ -95,16 +95,18 @@ export const List = ({
   const selectedBulkItemsCount = useAtomValue(selectedBulkItemsCountAtom);
   const setSelectedBulkItems = useSetAtom(setSelectedBulkItemsAtom);
   const selectedBulkItems = useAtomValue(selectedBulkItemsAtom);
-  const hasSelectedAllItems = selectedBulkItems.includes(ALL_SELECTED_KEY);
-  const hasSelectedItems = selectedBulkItems.length > 0;
+
+  const hasSelectedAllItems = isSelectingAllItems(selectedBulkItems);
+
+  const hasSelectedItems = selectedBulkItemsCount > 0;
 
   /**
    * We can select all the incoming items and we can add ALL_SELECTED_KEY
    * in the selected items. We check in backend for this ALL_SELECTED_KEY, if it is selected
    * then we do operation on all items of organization
    */
   function handleSelectAllItems() {
-    setSelectedBulkItems([...items.map((item) => item.id), ALL_SELECTED_KEY]);
+    setSelectedBulkItems([...items, { id: ALL_SELECTED_KEY }]);
```

Comment:
Okey this will work but I suspect it can create some issues. Can we please use the `prev` approach and just merge `{ id: ALL_SELECTED_KEY }` with the already existing state?